pkg/kube/podutil/podutil.go

```diff
@@ -35,3 +42,30 @@ func GetPodCondition(status *v1.PodStatus, conditionType v1.PodConditionType) (i
 	}
 	return -1, nil
 }
+
+// GetPod waits for a pod with the specified label to be ready, then returns it
+// if no pod is ready, it checks every second until a pod is ready
+func GetPod(namespace, labelKey, labelValue string, clientset kubernetes.Interface) *v1.Pod {
```

Comment:
Would it be better to accept a label selector instead? e.g. https://github.com/kubernetes/helm/blob/48e703997016f3edeb4f0b90e6cfdb3456ce3db0/pkg/helm/portforwarder/portforwarder.go#L58-L59

Reply:
Yes, it makes sense as will also add a label with the build ID later on.